crates/bevy_pbr/src/decal/clustered.rs

```diff
@@ -97,6 +98,72 @@ pub struct ClusteredDecal {
     pub tag: u32,
 }
 
+/// Cubemap layout defines the order of images in a packed cubemap image.
+#[derive(Default, Reflect, Debug, Clone, Copy)]
+pub enum CubemapLayout {
+    /// layout in a vertical cross format
+    ///    +y
+    /// -x -z +x
+    ///    -y
+    ///    +z
+    #[default]
+    CrossVertical = 0,
+    /// layout in a horizontal cross format
+    ///    +y
+    /// -x -z +x +z
+    ///    -y
+    CrossHorizontal = 1,
+    /// layout in a vertical sequence
+    ///   +x
+    ///   -y
+    ///   +y
+    ///   -y
+    ///   -z
+    ///   +z
+    SequenceVertical = 2,
+    /// layout in a horizontal sequence
+    /// +x -y +y -y -z +z
+    SequenceHorizontal = 3,
+}
+
+/// Add to a [`PointLight`] to add a light cookie effect.
+/// A light cookie is a texture mask applied to a light source to modulate its intensity,  
+/// simulating patterns like window shadows, gobo effects, or soft falloffs.
+#[derive(Clone, Component, Debug, Reflect)]
+#[reflect(Component, Debug)]
+#[require(PointLight)]
+pub struct PointLightCookie {
+    /// The cookie image. Only the R channel is read.
+    pub image: Handle<Image>,
+    /// The cubemap layout. The image should be a packed cubemap in one of the formats described by the [`CubemapLayout`] enum.
+    pub cubemap_layout: CubemapLayout,
+}
+
+/// Add to a [`SpotLight`] to add a light cookie effect.
+/// A light cookie is a texture mask applied to a light source to modulate its intensity,  
+/// simulating patterns like window shadows, gobo effects, or soft falloffs.
+#[derive(Clone, Component, Debug, Reflect)]
+#[reflect(Component, Debug)]
+#[require(SpotLight)]
+pub struct SpotLightCookie {
```

Comment:
3D packages don't agree on the term for this concept:

* Unreal: light functions
* Unity: light cookie
* Godot: light projector
* Blender: setting a texture as the light color

Of all of these terms, I find "light cookie" the least helpful. I like "light texture" or "light projector". "light function" would be too generic for the feature as implemented.

Reply:
light texture is the clearest to me as a rendering noob. We should doc alias the synonyms of course.

Reply:
It is called a cookie in theatre too apparently, from [cucoloris](https://en.m.wikipedia.org/wiki/Cucoloris). Light texture seems unambiguous though.

Reply:
renamed to `PointLightTexture` etc

Reply:
Good call. Thanks!